src/site/antora/modules/ROOT/pages/manual/appenders.adoc

```diff
@@ -234,38 +234,38 @@ This setting might be useful in multi-application environments.
 Some application servers modify `System.out` and `System.err` to always point to the currently running application.
 ====
 
+[#ConsoleAppender-jansi]
 `JANSI`::
-If the application is running on Windows and the
-https://fusesource.github.io/jansi/[JANSI library]
-is available, the Console appender will use JANSI to emulate ANSI sequence support.
-This mode can be enabled by setting the
-xref:manual/systemproperties.adoc#log4j.console.jansiEnabled[`log4j.console.jansiEnabled`]
-configuration attribute to `true`.
+If the
+xref:components.adoc#log4j-jansi[`log4j-jansi` extension]
+is available, the Console appender will use the
+https://fusesource.github.io/jansi/[JAnsi library]
+to handle ANSI escapes.
+The library:
 +
-Additional runtime dependencies are required to use JANSI:
+--
+* Implement ANSI escape handling on old Windows consoles.
+* Strips ANSI escape codes if process output is being redirected and not attached to a terminal
+--
 +
-[tabs]
+See https://github.com/fusesource/jansi?tab=readme-ov-file#features[Features of JAnsi] for more details.
++
+[TIP]
+====
+Support for ANSI escape sequences was added to Windows 10 console around 2017.
```

Comment:
This translates to Log4j 3 will cater for **Windows 10 users that haven't updated their systems since 2017**. Instead of this, shouldn't we remove JAnsi support completely and direct such users, if they indeed do exist, to Log4j 2 instead?

/cc @garydgregory @rgoers

Reply:
Does this mean that my current Windows 10 OS is supported? It is "current" since it receives regular updates and shows version "10.0.19045.4780" when I run "ver".

Reply:
Let me expand on that. The situation on a modern Windows 10+ is the following:

- All terminal windows are capable of interpreting ANSI escapes.
- The `cmd.exe` terminal by default has ANSI escape support switched off. You need to create the `HKEY_CURRENT_USER\Console\VirtualTerminalLevel` registry key and set it to `0x1` (as DWORD).
- Other applications have the feature switched on by default (I believe PowerShell and certainly the standalone Windows Terminal app).

Note also that if `VirtualTerminalLevel` is not enabled, a developer can enable it for his application by calling [`AnsiConsole.systemInstall()`](https://www.javadoc.io/doc/org.jline/jline/latest/org/jline/jansi/AnsiConsole.html).

Therefore:

- For a developer, `log4j-jansi` is useless. He can just call `AnsiConsole.systemInstall()` in his code.
- For a user, `log4j-jansi` can be used as an alternative to setting `VirtualTerminalLevel` to `0x1`.

All in all, I think we can drop `log4j-jansi` in `3.x`.